### ISSUE TITLE
error[E0432]: unresolved import `futures_macro::proc_macro_hack_join`

### ISSUE THREAD

**Issue by GopherJ** [rust-lang/futures-rs#2150]
```
error[E0432]: unresolved import `futures_macro::proc_macro_hack_join`
  --> /home/cheng/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.4/src/async_await/join_mod.rs:78:13
   |
78 |     pub use futures_macro::join;
   |             ^^^^^^^^^^^^^^^^^^^ no `proc_macro_hack_join` in the root

error[E0432]: unresolved import `futures_macro::proc_macro_hack_try_join`
  --> /home/cheng/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.4/src/async_await/join_mod.rs:81:13
   |
81 |     pub use futures_macro::try_join;
   |             ^^^^^^^^^^^^^^^^^^^^^^^ no `proc_macro_hack_try_join` in the root

error[E0432]: unresolved import `futures_macro::proc_macro_hack_select`
   --> /home/cheng/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.4/src/async_await/select_mod.rs:316:13
    |
316 |     pub use futures_macro::select;
    |             ^^^^^^^^^^^^^^^^^^^^^ no `proc_macro_hack_select` in the root

error[E0432]: unresolved import `futures_macro::proc_macro_hack_select_biased`
   --> /home/cheng/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.4/src/async_await/select_mod.rs:319:13
    |
319 |     pub use futures_macro::select_biased;
    |             ^^^^^^^^^^^^^^^-------------
    |             |              |
    |             |              help: a similar name exists in the module: `proc_macro_hack_select_internal`
    |             no `proc_macro_hack_select_biased` in the root

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0432`.
error: could not compile `futures-util`.

To learn more, run the command again with --verbose.

```

**Comment by GopherJ** [comment]
I think `future_util` cannot compile on: `rustc 1.45.0-nightly (a08c47310 2020-05-07)`

**Comment by GopherJ** [comment]
closed as `v0.3.5` fixed this.

**Comment by taiki-e** [comment]
This is probably because the futures-util and futures-macro versions are different.
https://github.com/rust-lang/futures-rs/pull/2124#issue-401919919
> Note that proc-macro is renamed to *_internal.
IIRC, it is not expected to use futures-macro directly, so this is probably fine...
(Maybe using = requirements would help to avoid some problems like #2061.)
